### PR TITLE
Fixes inconsistent rooms in home behaviour

### LINKED ISSUE #6510: Rooms in spaces unexpected change behaviour when visiting `settings -> preferences`
### Steps to reproduce

1. Have an account with a space and a room in that space
2. With a fresh install, log in and notice the room within the space is showing in `home`
3. Open the settings -> preference screen (don't change any toggles)
4. Quit and relaunch the app, notice the room from the space is no longer showing in `home`

#### What did you expect?
- For rooms within a space to not show in `home` by default
- For visiting a screen to not have unintended side effects

#### What happened instead?
- Rooms are shown in `home` by default until the settings -> preference page is opened 

### Your phone model

_No response_

### Operating system version

_No response_

### Application version and app store

_No response_

### Homeserver

_No response_

### Will you send logs?

No

### Are you willing to provide a PR?

Yes

### PR DESCRIPTION
**Pull request by ouchadam**
## Type of change

- [ ] Feature
- [x] Bugfix
- [ ] Technical
- [ ] Other :

## Content

- Only making use of the legacy `SETTINGS_LABS_SPACES_HOME_AS_ORPHAN` if it exists and the new key does not
- Manually deletes the legacy `SETTINGS_LABS_SPACES_HOME_AS_ORPHAN` after migrating to `SETTINGS_PREF_SPACE_SHOW_ALL_ROOM_IN_HOME`

## Motivation and context

Fixes #6510 Inconsistent _rooms in home_ behaviour

The app incorrectly returns the `Show all rooms in home` preference as enabled when it's meant to be disabled by default. This is caused by the migration logic from the previous flag not taking into account if the preference existed, causing us to always default to enabled. 

Will also fix some flakiness in the UI Tests!

## Screenshots / GIFs

|Before|After|
|-|-|
|before-space-rooms|after-spaces-room|
 

## Tests

- Have an account with a space and a room in that space
- With a fresh install, log in and notice the room within the space is showing in home
- Open the settings -> preference screen (don't change any toggles)
- Quit and relaunch the app, notice the room from the space is no longer showing in home

## Tested devices

- [x] Physical
- [ ] Emulator
- OS version(s): 29

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/settings/VectorPreferences.kt [hunk 1]
```diff
@@ -1020,20 +1020,32 @@ class VectorPreferences @Inject constructor(
...
+        val showAllRoomsInHome = !labsSpacesOnlyOrphansInHome()
+        defaultPrefs.edit {
+            putBoolean(SETTINGS_PREF_SPACE_SHOW_ALL_ROOM_IN_HOME, showAllRoomsInHome)
+            remove(SETTINGS_LABS_SPACES_HOME_AS_ORPHAN)
+        }
+        return showAllRoomsInHome
+    }
+
+    private fun labsSpacesOnlyOrphansInHome(): Boolean {
+        return defaultPrefs.getBoolean(SETTINGS_LABS_SPACES_HOME_AS_ORPHAN, false)
```
Comment (onurays) [inline vector/src/main/java/im/vector/app/features/settings/VectorPreferences.kt:1048]: Do we have to live with this or can we remove this flag in the next releases?

**Review by onurays — APPROVED** [review]
LGTM! Just added a question.

File: vector/src/main/java/im/vector/app/features/settings/VectorPreferences.kt (hunk 1 above)
Comment (ouchadam) [reply vector/src/main/java/im/vector/app/features/settings/VectorPreferences.kt:1048]: good question! it looks like the orphan flag was added [14 months](https://github.com/vector-im/element-android/pull/3312) ago then replaced [2 months](https://github.com/vector-im/element-android/pull/3765) later 

as it's been over a year I would lean towards removing the migration entirely! what do you think?

Comment (onurays) [reply vector/src/main/java/im/vector/app/features/settings/VectorPreferences.kt:1048]: I totally agree. They (if there are any users) shouldn't be so surprised.

Comment (ouchadam) [reply vector/src/main/java/im/vector/app/features/settings/VectorPreferences.kt]: removed! https://github.com/vector-im/element-android/pull/6511/commits/6266b2019b01c725299a5d0c6cd02b5249a686ad

Comment (onurays) [reply vector/src/main/java/im/vector/app/features/settings/VectorPreferences.kt]: LGTM! Feel free to merge.

**Pull request merged**